roqoqo-qryd/src/qryd_devices.rs

```diff
@@ -843,3 +845,371 @@ impl Device for FirstDevice {
+    ) {
+        let layout_name = layout_name.unwrap_or_else(|| self.current_layout.clone());
+
+        if let Some(info) = self.layout_register.get_mut(&layout_name) {
+            let sqt = &mut info.tweezer_three_qubit_gate_times;
+            if let Some(present_hm) = sqt.get_mut(hqslang) {
+                present_hm.insert((tweezer0, tweezer1, tweezer2), gate_time);
+            } else {
+                let mut hm = HashMap::new();
+                hm.insert((tweezer0, tweezer1, tweezer2), gate_time);
+                sqt.insert(hqslang.to_string(), hm);
+            }
+        }
+    }
+
+    /// Set the time of a multi-qubit gate for a list of tweezers in a given Layout.
+    ///
+    /// # Arguments
+    ///
+    /// * `hqslang` - The hqslang name of a multi-qubit gate.
+    /// * `tweezers` - The list of tweezer indexes.
+    /// * `gate_time` - The the gate time for the given gate.
+    /// * `layout_name` - The name of the Layout to apply the gate time in. Defaults to the current Layout.
+    ///
+    pub fn set_tweezer_multi_qubit_gate_time(
+        &mut self,
+        hqslang: &str,
+        tweezers: &[usize],
+        gate_time: f64,
+        layout_name: Option<String>,
+    ) {
+        let layout_name = layout_name.unwrap_or_else(|| self.current_layout.clone());
+
+        if let Some(info) = self.layout_register.get_mut(&layout_name) {
+            let sqt = &mut info.tweezer_multi_qubit_gate_times;
+            if let Some(present_hm) = sqt.get_mut(hqslang) {
+                present_hm.insert(tweezers.to_vec(), gate_time);
+            } else {
+                let mut hm = HashMap::new();
+                hm.insert(tweezers.to_vec(), gate_time);
+                sqt.insert(hqslang.to_string(), hm);
+            }
+        }
+    }
+
+    /// Get the tweezer identifier of the given qubit.
+    ///
+    /// # Arguments
+    ///
+    /// * `qubit` - The input qubit identifier.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(usize)` - The tweezer identifier relative to the given qubit.
+    /// * `Err(RoqoqoBackendError)` - If the qubit idetifier is not related to any tweezer.
+    ///
+    pub fn get_tweezer_from_qubit(&self, qubit: &usize) -> Result<usize, RoqoqoBackendError> {
+        if let Some(x) = self.qubit_to_tweezer.get(qubit) {
+            return Ok(*x);
+        }
+        Err(RoqoqoBackendError::GenericError {
+            msg: "The given qubit is not present in the Layout.".to_string(),
+        })
+    }
+
+    fn get_current_layout_info(&self) -> &TweezerLayoutInfo {
+        self.layout_register
+            .get(&self.current_layout)
+            .expect("Unexpectedly did not find current layout. Bug in roqoqo-qryd")
+    }
+}
+
+impl Device for ExperimentalDevice {
+    fn single_qubit_gate_time(&self, hqslang: &str, qubit: &usize) -> Option<f64> {
+        let tweezer_layout_info = self.get_current_layout_info();
+        let mapped_qubit = self.get_tweezer_from_qubit(qubit).unwrap();
```

Comment:
This should just return none if no tweezer position for the qubit can be found, it does not exist.